### PR TITLE
Fix module support

### PR DESCRIPTION
**Pull request by talbright**
I didn't follow the rules of ~~fight~~ mod club, resulting in the `+incompatible` marker for users of this library. 😠 

As a result of Semantic Import Versioning,  _code opting in to Go modules must comply with these rules_:
* Follow semver.
* **If the module is version v2 or higher, the major version of the module must be included as a /vN at the end of the module paths used in go.mod files (e.g., module github.com/my/mod/v2, require github.com/my/mod/v2 v2.0.0) and in the package import path (e.g., import "github.com/my/mod/v2/mypkg").**
* If the module is version v0 or v1, do not include the major version in either the module path or the import path.

### with go modules 

1. Create a new branch based on this one in this repository, not a fork. Note SHA1.
2. Create a new project    
```bash
$ mkdir myproj && cd myproj
```
3. Initialize the new project
```bash
$ go mod init
```
4. Create main
```go
package main

import (
	"context"
	"flag"
	"fmt"
	"log"

	heroku "github.com/heroku/heroku-go/v5"
)

var (
	username = flag.String("username", "", "api username")
	password = flag.String("password", "", "api password")
)

func main() {
	log.SetFlags(0)
	flag.Parse()

	heroku.DefaultTransport.Username = *username
	heroku.DefaultTransport.Password = *password

	h := heroku.NewService(heroku.DefaultClient)
	addons, err := h.AddOnList(context.TODO(), &heroku.ListRange{Field: "name"})
	if err != nil {
		log.Fatal(err)
	}
	for _, addon := range addons {
		fmt.Println(addon.Name)
	}
}
```
5.  Get the dependency
```bash
$ go get -u github.com/heroku/heroku-go/v5@sha1
```
6. Build the project
```bash
$ go build
```
7. Verify dependencies in `go.mod`

Follow similar procedure for the Heroku Terraform provider.

### without go modules 

1. Grab a pre-module version of golang
```bash
$ docker pull golang:1.9-alpine
```
2.  Fire up the image
```bash
$ docker run -it golang:1.9-alpine /bin/sh
```
3. Create main
```go
package main

import (
	"context"
	"flag"
	"fmt"
	"log"

	heroku "github.com/talbright/heroku-go/v5"
)

var (
	username = flag.String("username", "", "api username")
	password = flag.String("password", "", "api password")
)

func main() {
	log.SetFlags(0)
	flag.Parse()

	heroku.DefaultTransport.Username = *username
	heroku.DefaultTransport.Password = *password

	h := heroku.NewService(heroku.DefaultClient)
	addons, err := h.AddOnList(context.TODO(), &heroku.ListRange{Field: "name"})
	if err != nil {
		log.Fatal(err)
	}
	for _, addon := range addons {
		fmt.Println(addon.Name)
	}
}
```
5.  Get the dependency. 
```bash
$ go get github.com/talbright/heroku-go/v5
```
6. Verify the project builds without module support
```bash
$ go build
```

### CODE REVIEW

File: v5/heroku.go [hunk 1]
```diff
@@ -8,24 +8,25 @@
...
 	"runtime"
 	"strings"
 	"time"
+
+	"github.com/google/go-querystring/query"
 )
 
 const (
-	Version          = "v3"
+	Version          = "v5"
```
Comment (talbright) [inline v5/heroku.go]: Heads up here...

**Comment by talbright** [comment]
@mars ready for review

**Comment by mars** [comment]
Looks good @talbright! I'll have a chance to review thoroughly later this week; will test by using this branch with the Terraform Provider.

**Comment by talbright** [comment]
No worries. Updated directions for testing w/earlier versions of go that don't support go modules.

**Comment by mars** [comment]
This modules stuff has been challenging to validate in the Terraform Provider 😅

I followed the **Testing with Go modules** instructions above, but the commit in my local branch is not found:

```bash
$  go get github.com/heroku/heroku-go/v5@0efb7b0
go: finding github.com/heroku/heroku-go/v5 0efb7b0
go: finding github.com/heroku/heroku-go 0efb7b0
go get github.com/heroku/heroku-go/v5@0efb7b0: unknown revision 0efb7b0
```

Trying to tidy-up the module specs:

```bash
$  go mod tidy
go: finding github.com/heroku/heroku-go/v5 latest
go: downloading github.com/heroku/heroku-go/v5 v5.0.0-20190503184120-a7e195a6f672
```

…but failing there. `heroku/heroku-go/v5` is not actually downloaded. The module cannot be found.

To make the module retrievable, I decided to create a Beta release tag on the GitHub repo. I pushed the PR branch to [talbright-talbright/fix-mod-compat](https://github.com/heroku/heroku-go/tree/talbright-talbright/fix-mod-compat) and [released it as `v5.0.0-beta.1`](https://github.com/heroku/heroku-go/releases/tag/v5.0.0-beta.1).

Then, in the Terraform Provider I fetched the module:

```bash
$  go get github.com/heroku/heroku-go/v5@v5.0.0-beta.1
$  go mod tidy
$  go mod vendor
```

Installation success!

I created a [pull request on the Provider](https://github.com/terraform-providers/terraform-provider-heroku/pull/196) to Upgrade heroku-go API client to `v5.0.0-beta.1`. Its unit tests pass and acceptance tests are running in [Hashicorp CI](https://ci-oss.hashicorp.engineering/viewLog.html?buildId=57320&tab=buildResultsDiv&buildTypeId=HerokuProvider_ProviderHeroku). I'll update when that completes ⏳

**Comment by talbright** [comment]
> Create a new branch based on this one in this repository, **not a fork**. Note SHA1.
You have to create a branch in this repo -- then it should resolve.

**Comment by mars** [comment]
Yes, that's basically what I figured out. I didn't realize "this repository" literally meant this GitHub remote.

**Pull request merged**